test/relations.test.js

```diff
@@ -109,6 +122,25 @@ describe('relations', function () {
         });
       });
 
+      it('should create a batch of records on scope with promises', function (done) {
```

Comment:
Is it a good idea to support promise-enabled batch create, considering the issue we have discussed in the previous PR?

Since the callback-based version calls `async.map` and thus the promise-variant does not require significant changes and simply preserves all quirks of the callback-based version, I am ok with keeping this in, just wanted to check.